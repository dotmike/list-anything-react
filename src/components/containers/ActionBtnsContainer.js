import React from "react";
import PropTypes from "prop-types";
import { Button, Popup, Confirm } from "semantic-ui-react";

class ActionBtnsContainer extends React.Component {
  state = { showConfirm: false };

  handleCancel = () => this.setState({ showConfirm: false });

  showConfirm = () => this.setState({ showConfirm: true });

  render() {
    const { onMenu, onAdd, onEdit, onDelete, style } = this.props;
    const { showConfirm } = this.state;
    return (
      <div style={style}>
        {(!!onMenu || !!onAdd) && (
          <Button.Group size="medium" icon>
            {!!onMenu && (
              <Popup
                trigger={<Button icon="sidebar" onClick={onMenu} />}
                content="Side Menu"
              />
            )}
            {!!onAdd && (
              <Popup
                trigger={<Button icon="plus" color="green" onClick={onAdd} />}
                content="Add"
              />
            )}
          </Button.Group>
        )}
        {(!!onEdit || !!onDelete) && (
          <Button.Group size="medium" icon>
            {!!onEdit && (
              <Popup
                trigger={<Button icon="edit" color="blue" onClick={onEdit} />}
                content="Edit"
              />
            )}
            {!!onDelete && (
              <Popup
                trigger={
                  <Button
                    icon="delete"
                    color="red"
                    onClick={this.showConfirm}
                  />
                }
                content="Delete"
              />
            )}
            {!!onDelete && (
              <Confirm
                open={showConfirm}
                onCancel={this.handleCancel}
                onConfirm={onDelete}
              />
            )}
          </Button.Group>
        )}
      </div>
    );
  }
}

ActionBtnsContainer.propTypes = {
  // ownProps
  onMenu: PropTypes.func,
  onAdd: PropTypes.func,
  onEdit: PropTypes.func,
  onDelete: PropTypes.func,
  style: PropTypes.objectOf(PropTypes.string)
};

ActionBtnsContainer.defaultProps = {
  onMenu: null,
  onAdd: null,
  onEdit: null,
  onDelete: null,
  style: {}
};

export default ActionBtnsContainer;
