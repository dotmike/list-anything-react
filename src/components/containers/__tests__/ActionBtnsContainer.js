import React from "react";
import { shallow } from "enzyme";

import ActionBtnsContainer from "../ActionBtnsContainer";

const defaultProps = {
  onMenu: () => {},
  onAdd: () => {},
  onEdit: () => {},
  onDelete: () => {}
};

describe("ActionBtnsContainer", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<ActionBtnsContainer {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
