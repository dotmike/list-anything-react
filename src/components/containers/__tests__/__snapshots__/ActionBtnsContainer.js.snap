// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionBtnsContainer renders correctly 1`] = `
<div
  style={Object {}}
>
  <ButtonGroup
    icon={true}
    size="medium"
  >
    <Popup
      content="Side Menu"
      on="hover"
      position="top left"
      trigger={
        <Button
          as="button"
          icon="sidebar"
          onClick={[Function]}
        />
      }
    />
    <Popup
      content="Add"
      on="hover"
      position="top left"
      trigger={
        <Button
          as="button"
          color="green"
          icon="plus"
          onClick={[Function]}
        />
      }
    />
  </ButtonGroup>
  <ButtonGroup
    icon={true}
    size="medium"
  >
    <Popup
      content="Edit"
      on="hover"
      position="top left"
      trigger={
        <Button
          as="button"
          color="blue"
          icon="edit"
          onClick={[Function]}
        />
      }
    />
    <Popup
      content="Delete"
      on="hover"
      position="top left"
      trigger={
        <Button
          as="button"
          color="red"
          icon="delete"
          onClick={[Function]}
        />
      }
    />
    <Confirm
      cancelButton="Cancel"
      confirmButton="OK"
      content="Are you sure?"
      onCancel={[Function]}
      onConfirm={[Function]}
      open={false}
    />
  </ButtonGroup>
</div>
`;
